mantidproject/mantid
# Update to EstimateMuonAsymmetryFromCounts

The calculation of the muon asymmetry is different between the doc and code. This difference is from replacing the denominator with an integral, then estimating it with the trapezium rule. 

If the `startX` is the same as a bin edge, then the bin is missing (which is wrong) from the calculation.

General update to the input/output and integration into Muon analysis

The code was missing a subtraction for the two edge cases. 

Note that the changes to muon process test are because it uses this algorithm to produce its result. These changes have also impacted CaclulcateMuonAsymmetry, but in issue #22490 the algorithm is getting a big rewrite.

**To test:**
Open Muon Analysis and load `musr 62260`
The `MuonAnalysisTFNormalizations` will contain the normalization value and the workspace name
Go to the data analysis tab
Add a flat background and a GaussOsc function
Change the frequency value to 1.3
Do a fit
The flat background should be small but non zero

Remove the flat background
Do a TF Asymmetry fit
The `MuonAnalysisTFNormalizations` table should be updated

Go to the settings tab and turn on Multiple fitting mode
Go to data analysis
Check the table now contains the information for all 4 groups

Run the below script and you will get the same result for all 3 outputs -> the bin edges are included correctly.
```
import math
import numpy as np

tab = CreateEmptyTableWorkspace()
tab.addColumn('double', 'norm')
tab.addColumn('str', 'name')
tab.addColumn('str', 'method')

tab.addRow([1.,"Run;;Group;;a;;Asym;;#1","Estimate"])

y = [100, 150, 50, 10, 5]
x = [1,2,3,4,5,6]
input = CreateWorkspace(x,y)
run = input.getRun()
run.addProperty("goodfrm","10","None",True)
output=EstimateMuonAsymmetryFromCounts(InputWorkspace=input,NormalizationTable=tab,spectra=0,StartX=1,EndX=5)
print("Asymmetry:  {}".format(['{0:.2f}'.format(value) for value in output.readY(0)]))

output=EstimateMuonAsymmetryFromCounts(InputWorkspace=input,NormalisationTable=tab,spectra=0,StartX=.9,EndX=5.)
print("Asymmetry:  {}".format(['{0:.2f}'.format(value) for value in output.readY(0)]))

output=EstimateMuonAsymmetryFromCounts(InputWorkspace=input,NormalisationTable=tab,spectra=0,StartX=1,EndX=5.1)
print("Asymmetry:  {}".format(['{0:.2f}'.format(value) for value in output.readY(0)]))
```




Fixes #22464. 
Fixes #22465 
Fixes #22471
**Release Notes** 

In release notes
---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
